### DISCUSSION TITLE
Lesson 5: cannot import name 'SimpleStorage' from 'brownie'

### DISCUSSION THREAD

**Discussion by s1n7** [smartcontractkit/full-blockchain-solidity-course-py#185] [Q&A]
Hey Guys! :)
I'm having issues deploying the deploy.py file. The issue arrised when I tried to import SimpleStorage to the deploy.py script. There is an error occurring saying: "cannot import name 'SimpleStorage' from 'brownie' ". However, the testing approach works fine, importing SimpleStorage there seems to be no problem at all.

**Comment by PatrickAlphaC** [comment]
Could you add:

1. What command you're running
2. The full error
3. The script code
4. THe contract code
5. An image of what your text editor looks like

**Reply by s1n7** [reply]
1. command I'm running: `brownie run project/scripts/deploy.py` The command looks like this because I had to create a virtual environment and had to create an empty folder (project) because somehow pipx does not install brownie. 
2. full error:
![image](https://user-images.githubusercontent.com/65351091/137638393-6c4e7124-c6a2-4cb5-a82f-19a29d11b62b.png)

3. script code from deploy.py: 
`from brownie import SimpleStorage, accounts, network, config
import os

from brownie.network import account

def deploy_simple_storage():
    #work with local accounts:
    account = accounts[0]
    simple_storage = SimpleStorage.deploy({'from': account})
    stored_value = simple_storage.retrieve()
    print(stored_value)


def get_account():
    if(network.show_active() == "development"):
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])

def main():
    deploy_simple_storage()`

4. SimpleStorage.sol code:
`// SPDX-License-Identifier: MIT

pragma solidity ^0.6.0;

contract SimpleStorage {
    // public defines the visibility, default is "internal" as visibility
    uint256 public favoriteNumber;

    bool favoriteBool;

    // ways to define new types in solidity
    struct People {
        uint256 favoriteNumber;
        string name;
    }

    //dynamic array: can change size
    People[] public people;
    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        favoriteNumber = _favoriteNumber;
    }

    // view, pure
    function retrieve() public view returns (uint256) {
        return favoriteNumber;
    }

    //memory: data will only be stored during the execution of the function
    //storage: data will persist after the function executes
    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        people.push(People(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}
`
5. text editor: 
![image](https://user-images.githubusercontent.com/65351091/137638488-df018b13-ae0b-4101-b006-f990cb05c228.png)

**Reply by cromewar** [reply]
I faced a similar problem before, try to do this instead:

Create a Folder (for the project), inside of if place another to (One for the virtualenv and another for the code itself) Something like this:

Project folder
| - dev (for code)
| - virtualenv

then, delete the `build` folder and compile again:

`brownie compile`

Finally try again running the script, hope this helps.

Note: remember to install eth-brownie instead of just brownie.
Note: to, don't forget to activate the environment, as it is now outside the code folder activate it using `../restOfThePath`.

**Reply by s1n7** [reply]
It worked now, thank you! :)

**Reply by cromewar** [reply]
No problem, I'm glad it helped :)

**Reply by PatrickAlphaC** [reply]
bravo!
